src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -96,13 +96,16 @@ public void ValidateConfig()
             IFileSystem fileSystem,
             ILogger logger)
         {
-            // Connection string can't be null or empty
-            if (string.IsNullOrWhiteSpace(runtimeConfig.DataSource.ConnectionString))
+            foreach (DataSource dataSource in runtimeConfig.DataSourceNameToDataSource.Values)
             {
-                throw new DataApiBuilderException(
-                    message: DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE,
-                    statusCode: HttpStatusCode.ServiceUnavailable,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
+                // Connection string can't be null or empty
+                if (string.IsNullOrEmpty(dataSource.ConnectionString))
+                {
+                    throw new DataApiBuilderException(
```

Comment:
Why change from `IsNullOrWhiteSpace()`? is there a valid scenario where connectionstring is all whitespace?

Reply:
Fixed. Was rewriting this code and typically always used isnullorempty. moved back to isnullorwhitespace.